homeassistant/components/group/__init__.py

```diff
@@ -629,57 +696,40 @@ def _async_update_group_state(self, tr_state=None):
         This method must be run in the event loop.
         """
         # To store current states of group entities. Might not be needed.
-        states = None
-        gr_state = self._state
-        gr_on = self.group_on
-        gr_off = self.group_off
-
-        # We have not determined type of group yet
-        if gr_on is None:
-            if tr_state is None:
-                states = self._tracking_states
-
-                for state in states:
-                    gr_on, gr_off = _get_group_on_off(state.state)
-                    if gr_on is not None:
-                        break
-            else:
-                gr_on, gr_off = _get_group_on_off(tr_state.state)
-
-            if gr_on is not None:
-                self.group_on, self.group_off = gr_on, gr_off
+        if tr_state:
+            self._see_state(tr_state)
 
-        # We cannot determine state of the group
-        if gr_on is None:
+        if not self._on_off:
             return
 
-        if tr_state is None or (
-            (gr_state == gr_on and tr_state.state == gr_off)
-            or (gr_state == gr_off and tr_state.state == gr_on)
-            or tr_state.state not in (gr_on, gr_off)
-        ):
-            if states is None:
-                states = self._tracking_states
-
-            if self.mode(state.state == gr_on for state in states):
-                self._state = gr_on
-            else:
-                self._state = gr_off
-
-        elif tr_state.state in (gr_on, gr_off):
-            self._state = tr_state.state
-
         if (
             tr_state is None
             or self._assumed_state
             and not tr_state.attributes.get(ATTR_ASSUMED_STATE)
         ):
-            if states is None:
-                states = self._tracking_states
-
-            self._assumed_state = self.mode(
-                state.attributes.get(ATTR_ASSUMED_STATE) for state in states
-            )
+            self._assumed_state = self.mode(self._assumed.values())
 
         elif tr_state.attributes.get(ATTR_ASSUMED_STATE):
             self._assumed_state = True
+
+        num_on_states = len(self._on_states)
+        # If all the entity domains we are tracking
+        # have the same on state we use this state
+        # and its hass.data[REG_KEY].on_off_mapping to off
+        if num_on_states == 1:
+            on_state = list(self._on_states)[0]
+        # If we do not have an on state for any domains
+        # we use STATE_UNKNOWN
+        elif num_on_states == 0:
+            self._state = STATE_UNKNOWN
```

Comment:
Our standard is to use `None` to represent unknown state. It will be converted by the base `Entity` class to `STATE_UNKNOWN`.

Reply:
Done in https://github.com/home-assistant/core/pull/40792